Sources/XRPLSwift/HDWallet/PrivateKey.swift

```diff
@@ -46,7 +46,10 @@ internal struct PrivateKey {
     internal var publicKey: Data {
         var _data = raw
         let ctx = secp256k1_context_create(UInt32(SECP256K1_CONTEXT_SIGN))!
-        let data = try! Data(SECP256K1.derivePublicKey(ctx: ctx, secretKey: _data.getPointer()).compressed)
+        let masterPublicKey = try! SECP256K1.derivePublicKey(ctx: ctx, secretKey: _data.getPointer())
```

Comment:
Does this work with ED25519 keys?

Reply:
The ED25519 crypto is in there and you can create a wallet and sign txs but for that function it might have been written for only SECP256K1.